### ISSUE TITLE
Upgrade to LLVM 18

### ISSUE THREAD

**Issue by zanieb** [astral-sh/python-build-standalone#233]
Looking into switching over to the official LLVM builds when upgrading to v18.

It doesn't look like they have binaries for the desired platforms on GitHub. I imagine we'll just install from the relevant system package managers or use download links from one of those?

e.g. in progress release on Homebrew:

- https://github.com/Homebrew/homebrew-core/pull/165206

**Comment by indygreg** [comment]
Yeah, it looks like the Linux x86-64 binaries published by LLVM only work on Ubuntu 18.04 or newer. Our Debian environment is too old and the official LLVM binaries aren't compatible.

I'm tempted to volunteer my services to the upstream LLVM project to get a _better_ LLVM toolchain published during releases. But that's off topic.

Over the weekend I did manage to publish LLVM 18.1.1 to https://github.com/indygreg/toolchain-tools/releases/tag/toolchain-bootstrap%2F20240316. In theory those should _just work_.

**Comment by zanieb** [comment]
What would you like to be better about the published LLVM toolchain?

Coming back to this because the Homebrew pull request finally finished. But it sounds like we need our own binaries for Debian and we won't be able to transition to the official builds?

**Comment by charliermarsh** [comment]
Should we close this out? It looks like we're now on LLVM 19.

**Comment by zanieb** [comment]
This is sort of tracking switching to the official LLVM distributions, perhaps we should just rename?
